airflow/providers/google/cloud/sensors/dataflow.py

```diff
@@ -187,4 +192,196 @@ def poke(self, context: dict) -> bool:
             location=self.location,
         )
 
+        if self.fail_on_terminal_state:
```

Comment:
This is inconsistent with: fetch_job_autoscaling_events_by_id, fetch_job_messages_by_id. Here we first query and then fail if job is in a terminal state. The other cases do the opposite. I think this option is better because, a job may reach to a terminal state after the initial query. Either way it will be good to use the same logic.

Reply:
Thanks @aaltay ! I applied your suggestion.